src/Symfony/Component/Serializer/Normalizer/DenormalizerInterface.php

```diff
@@ -30,11 +32,11 @@ interface DenormalizerInterface
      * Denormalizes data back into an object of the given class.
      *
      * @param mixed  $data    Data to restore
-     * @param string $type    The expected class to instantiate
-     * @param string $format  Format the given data was extracted from
+     * @param class-string<TObject> $type    The expected class to instantiate
+     * @param string|null $format  Format the given data was extracted from
```

Comment:
please mind fabbot for the CS, but please do not remove the `@param array  $context` that already exists - fabbot wants to remove them but we don't

Reply:
Fabbot action is still red, I hope I did it correctly?